pkg/kubernetessecrets/kubernetes_secrets_builder.go

```diff
@@ -67,15 +72,12 @@ func CreateKubernetesSecretFromItem(kubeClient kubernetesClient.Client, secretNa
 	if currentSecretType == "" {
 		currentSecretType = "Opaque"
 	}
-	if secretType == "" {
-		secretType = "Opaque"
-	}
 	if !reflect.DeepEqual(currentAnnotations, secretAnnotations) || !reflect.DeepEqual(currentLabels, labels) || !reflect.DeepEqual(currentSecretType, secretType) {
 		log.Info(fmt.Sprintf("Updating Secret %v at namespace '%v'", secret.Name, secret.Namespace))
 		currentSecret.ObjectMeta.Annotations = secretAnnotations
 		currentSecret.ObjectMeta.Labels = labels
 		currentSecret.Data = secret.Data
-		currentSecret.Type = KubernetesSecretTypes[secretType]
+		currentSecret.Type = corev1.SecretType(secretType)
```

Comment:
What happens when secretType is not a known secret type string by Kubernetes? Do we need validation or does Kubernetes do this in a later step?

Reply:
According to the docs any string can be a valid secret type. The ones defined in the kubernetes docs are handled differently by kubernetes, but any user can define their own types and handle them differently from other secret types.

[Quote](https://kubernetes.io/docs/concepts/configuration/secret/#secret-types):
> The type of a Secret is used to facilitate programmatic handling of different kinds of confidential data.
> [...]
> You can define and use your own Secret type by assigning a non-empty string as the type value for a Secret object. An empty string is treated as an Opaque type. Kubernetes doesn't impose any constraints on the type name. However, if you are using one of the builtin types, you must meet all the requirements defined for that type.